### PR TITLE
8253569: javax.xml.catalog.Catalog.matchURI() implementation should reset state variables

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Resets state of a Catalog instance on each matching call so that it can be reused. Adjusts CatalogResolver's resolve routine so that it continues to observes the PREFER feature in a resolution process. Without the adjustment, PreferFeatureTest would fail.

Mach5 XML tests passed: https://mach5.us.oracle.com/mdash/jobs/huizwang-open-20220329-0148-30563542

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253569](https://bugs.openjdk.java.net/browse/JDK-8253569): javax.xml.catalog.Catalog.matchURI() implementation should reset state variables

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to 14513d7bd7104ed86f0d25d6823aa577c1e2a54c
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8018/head:pull/8018` \
`$ git checkout pull/8018`

Update a local copy of the PR: \
`$ git checkout pull/8018` \
`$ git pull https://git.openjdk.java.net/jdk pull/8018/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8018`

View PR using the GUI difftool: \
`$ git pr show -t 8018`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8018.diff">https://git.openjdk.java.net/jdk/pull/8018.diff</a>

</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]
looks fine Joe

File: src/java.xml/share/classes/javax/xml/catalog/GroupEntry.java [hunk 1]
```diff
@@ -40,6 +40,12 @@
     static final int ATTRIBUTE_DEFFER = 1;
     static final int ATTRIBUTE_RESOLUTION = 2;
 
+    //Indicates whether this is the Catalog instance (vs a Catalog entry)
+    boolean isCatalogInstance = false;
```
Comment (naotoj) [inline src/java.xml/share/classes/javax/xml/catalog/GroupEntry.java:44]: Can we eliminate this instance variable, by checking whether the type is `Catalog` or not in `resetOnStart()`?

Comment (JoeWang-Java) [reply src/java.xml/share/classes/javax/xml/catalog/GroupEntry.java]: Yes, thanks!

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**